### PR TITLE
Upgrade cluster version for rhtap-nightly-devfile-compatibility to 4.13

### PR DESCRIPTION
**Pull request by thepetk**
## What this PR does?
This PR upgrades the ocp version used by the `periodic-ci-devfile-registry-main-v4.13-rhtap-nightly-devfile-compatibility` from `4.12` to `4.13` as right now all the runs for this check are failing with the trace:

```
NoClusters: No clusters in pool are ready to be claimed
```

With the upgrade proposed we will be able to claim a cluster of the 4.13 version.

### CODE REVIEW

**Review by Jdubrick — APPROVED** [review]
/lgtm

**Comment by thepetk** [comment]
@flacatus tried to upgrade the ocp version from 4.12 to 4.13 :)

File: ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml
Comment (michael-valdron) [inline ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml:26]: @thepetk This needs to be re-added to keep our slack notifications, job generation is currently removing this.

**Review by michael-valdron — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml
Comment (thepetk) [reply ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml]: @michael-valdron nice catch thanks :) I've re-added in the previous version of the PR but not the second time I ran the `make jobs`. I've re-added again!

File: ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml
Comment (michael-valdron) [inline ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml:19]:
```suggestion
      channel: '#team-has-github-alerts'
```

File: ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml
Comment (michael-valdron) [inline ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml:24]: Might of been a mix up with the reverted changes.

**Review by michael-valdron — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml
Comment (thepetk) [reply ci-operator/jobs/devfile/registry/devfile-registry-main-periodics.yaml]: Most probably. I've reverted back the original slack notification version.

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by michael-valdron** [comment]
> PR needs rebase.

cc: @thepetk

**Comment by thepetk** [comment]
> > PR needs rebase.
> 
> cc: @thepetk

done

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by thepetk** [comment]
@michael-valdron updated the jobs once again after the rebase. Needs again a lgtm label

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by michael-valdron** [comment]
@thepetk Looks like `github.com/devfile/registry/tests/rhtap` has some dependency failures we need to address:

```
 go: downloading github.com/redhat-appstudio/e2e-tests v0.0.0-20240304120539-8c8b8acfcc81
github.com/devfile/registry/tests/rhtap tested by
	github.com/devfile/registry/tests/rhtap.test imports
	github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes: module github.com/redhat-appstudio/e2e-tests@latest found (v0.0.0-20240304120539-8c8b8acfcc81), but does not contain package github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes
github.com/devfile/registry/tests/rhtap tested by
	github.com/devfile/registry/tests/rhtap.test imports
	github.com/redhat-appstudio/e2e-tests/pkg/framework imports
	github.com/redhat-appstudio/e2e-tests/pkg/clients/has imports
	github.com/redhat-appstudio/e2e-tests/pkg/utils/build imports
	k8s.io/cli-runtime/pkg/genericclioptions imports
	k8s.io/cli-runtime/pkg/resource imports
	k8s.io/client-go/openapi3: module k8s.io/client-go@latest found (v1.5.2, replaced by k8s.io/client-go@v0.25.5), but does not contain package k8s.io/client-go/openapi3
github.com/devfile/registry/tests/rhtap tested by
	github.com/devfile/registry/tests/rhtap.test imports
	github.com/redhat-appstudio/e2e-tests/pkg/framework imports
	github.com/redhat-appstudio/e2e-tests/pkg/clients/has imports
	github.com/redhat-appstudio/e2e-tests/pkg/utils/build imports
	github.com/openshift/oc/pkg/cli/image/extract imports
	k8s.io/kubectl/pkg/cmd/util imports
	k8s.io/kubectl/pkg/util/i18n imports
	github.com/chai2010/gettext-go/gettext: module github.com/chai2010/gettext-go@latest found (v1.0.2), but does not contain package github.com/chai2010/gettext-go/gettext
github.com/devfile/registry/tests/rhtap tested by
	github.com/devfile/registry/tests/rhtap.test imports
	github.com/redhat-appstudio/e2e-tests/pkg/framework imports
	github.com/redhat-appstudio/e2e-tests/pkg/clients/has imports
	github.com/redhat-appstudio/e2e-tests/pkg/utils/build imports
	k8s.io/cli-runtime/pkg/genericclioptions imports
	k8s.io/cli-runtime/pkg/resource tested by
	k8s.io/cli-runtime/pkg/resource.test imports
	k8s.io/apimachinery/pkg/util/dump: module k8s.io/apimachinery@latest found (v0.29.2, replaced by k8s.io/apimachinery@v0.25.5), but does not contain package k8s.io/apimachinery/pkg/util/dump
github.com/devfile/registry/tests/rhtap tested by
	github.com/devfile/registry/tests/rhtap.test imports
	github.com/redhat-appstudio/e2e-tests/pkg/framework imports
	github.com/redhat-appstudio/e2e-tests/pkg/clients/has imports
	github.com/redhat-appstudio/e2e-tests/pkg/utils/build imports
	k8s.io/cli-runtime/pkg/genericclioptions imports
	k8s.io/cli-runtime/pkg/resource tested by
	k8s.io/cli-runtime/pkg/resource.test imports
	k8s.io/client-go/openapi/openapitest: module k8s.io/client-go@latest found (v1.5.2, replaced by k8s.io/client-go@v0.25.5), but does not contain package k8s.io/client-go/openapi/openapitest
+ ginkgo run -p --timeout 2h tests/rhtap -- -samplesFile /tmp/tmp.Wq0PsGaa5n/registry/extraDevfileEntries.yaml
ginkgo run failed 
```

**Comment by michael-valdron** [comment]
@thepetk since its only `github.com/devfile/registry/tests/rhtap` that is failing rehearsals for dependency failures, we could merge these changes and address failures in https://github.com/devfile/registry/pull/310 and https://github.com/openshift/release/pull/49400. WDYT?

**Comment by michael-valdron** [comment]
@thepetk  **Revising my deleted comment**: Actually we should address the dependency failures before we merge these changes.

**Comment by thepetk** [comment]
> @thepetk **Revising my deleted comment**: Actually we should address the dependency failures before we merge these changes. I'll open an issue.

@michael-valdron yes I think is required as it might be related with the upgrade to 4.13

**Comment by thepetk** [comment]
> go: downloading github.com/redhat-appstudio/e2e-tests v0.0.0-20240304120539-8c8b8acfcc81
> github.com/devfile/registry/tests/rhtap tested by
> 	github.com/devfile/registry/tests/rhtap.test imports
> 	github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes: module github.com/redhat-appstudio/e2e-tests@latest found (v0.0.0-20240304120539-8c8b8acfcc81), but does not contain package github.com/redhat-appstudio/e2e-tests/pkg/apis/kubernetes

@michael-valdron I think the problem here is that we are using the `github.com/redhat-appstudio/e2e-tests v0.0.0-20240304120539-8c8b8acfcc81` while we should be using the `github.com/redhat-appstudio/e2e-tests v0.0.0-20230628083559-110126807738` (the next commit merges the `apis` packages with the `client` so it breaks the registry `rhtap` tests.

**Comment by michael-valdron** [comment]
fixes devfile/api#1468

**Comment by thepetk** [comment]
/pj-rehearse periodic-ci-devfile-registry-main-v4.13-rhtap-nightly-devfile-compatibility

**Comment by michael-valdron** [comment]
@thepetk Noticed `github.com/redhat-appstudio/e2e-tests v0.0.0-20240216111832-81d6508c8820` is on Go 1.20: https://github.com/redhat-appstudio/e2e-tests/blob/81d6508c88203bc5377c046700d67db459dd2968/go.mod#L3

Could this be why the nightly testing is still failing rehearsals? Maybe we should bump the registry test module to Go 1.20 instead, not sure how that would effect testing with HAS.

**Comment by thepetk** [comment]
/pj-rehearse periodic-ci-devfile-registry-main-v4.13-rhtap-nightly-devfile-compatibility

**Comment by thepetk** [comment]
Update regarding the new status failure here: https://github.com/devfile/api/issues/1468#issuecomment-1991665793

**Comment by thepetk** [comment]
/pj-rehearse periodic-ci-devfile-registry-main-v4.13-rhtap-nightly-devfile-compatibility

**Comment by thepetk** [comment]
/pj-rehearse periodic-ci-devfile-registry-main-v4.13-rhtap-nightly-devfile-compatibility

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by michael-valdron** [comment]
@thepetk Might need rebasing for lgtm to work.

**Comment by michael-valdron** [comment]
> @thepetk Might need rebasing for lgtm to work.

Nevermind, label was just slow to apply :smile:

**Pull request merged**
